packages/app/pages/checkout/[network]/[address].tsx

```diff
@@ -175,105 +195,142 @@ const CheckoutPage = ({
       });
     }
   }, [metadata, metadataHash]);
+  useEffect(() => {
+    if (qrFile)
+      QRCode.toCanvas(qrCanvasRef.current, qrFile).catch((e) =>
+        setError(e.message)
+      );
+  }, [qrFile, qrCanvasRef]);
   const onBuy = useCallback(() => {
     setError("");
     setLoading(true);
-    return (
-      web3.current.givenProvider.isConnected()
-        ? Promise.resolve()
-        : (web3.current.givenProvider.enable() as Promise<void>)
-    )
-      .then(() =>
-        Promise.all([
-          web3.current.eth.getAccounts(),
-          web3.current.eth.getChainId(),
-        ])
+    const contract = new web3.eth.Contract(
+      getAbiFromJson(passportJson),
+      address
+    );
+    return new Promise<string>((resolve, reject) =>
+      (
+        contract.methods.buy(
+          new BN(web3.utils.randomHex(32).replace(/^0x/, ""), "hex")
+        ) as ContractSendMethod
       )
-      .then(([accounts, chainId]) => {
-        if (Number(chainId) !== Number(networkIdByName[network])) {
-          setError(`Signed into the wrong network`);
-          return;
+        .send({
+          from: account,
+          value: web3.utils.toWei(price, "ether"),
+        })
+        .on("receipt", (receipt) => {
+          const tokenId =
+            (receipt.events?.["Purchase"]?.returnValues?.id as string) || "";
+          setLoading(false);
+          setSupply(supply - 1);
+          resolve(tokenId);
+        })
+        .on("error", reject)
+    )
+      .then((tokenId) => {
+        if (generateApplePass && tokenId) {
+          const signatureMessage = `Give Passports permission to generate an Apple Wallet Pass for token ${tokenId}`;
```

Comment:
Not very clear about the API but is there support for anything similar in Android? If yes, did we decide to do only Apple first?

Reply:
There wasn't support for Android when I started the PR, but there is now. Will be adding Android support via Google Pay on the next PR for this task